core/server/master/src/main/java/alluxio/master/metastore/rocks/RocksInodeStore.java

```diff
@@ -327,7 +344,7 @@ public void addChild(long parentId, String childName, Long childId) {
 
   @Override
   public void removeChild(long parentId, String name) {
-    try {
+    try (RocksSharedLockHandle lock = mRocksStore.checkAndAcquireSharedLock()){
       db().delete(mEdgesColumn.get(), mDisableWAL, RocksUtils.toByteArray(parentId, name));
```

Comment:
I'm wondering if we can move all the handling of the concurrency handling to inside the RocksStore object? For example here the lock check could take place inside the db().delete() call?
Maybe also then the mEdgesColumn lifecycle could be handled by the RocksStore as well?

Reply:
The `delete()` is in `RocksDB` class, which is in the RocksJava lib we don't have control over. So we can't easily move all these tailored logic into RocksStore trivially. However, there are ways we can achieve that, and instead of addressing in this PR, I will backlog this and address that as a design debt in 3.0/3.x. https://github.com/Alluxio/alluxio/issues/17271